### PR TITLE
Enable ImageLocalityPriority by default with integration tests

### PR DESCRIPTION
**Pull request by silveryfu**
**What this PR does / why we need it**:

This PR is a follow-up to [#63842](https://github.com/kubernetes/kubernetes/issues/63842). It moves the ImageLocalityPriority function to default priority functions of the default algorithm provider and adds integration tests for the updated scheduling policy.

- Compared to [#64662](https://github.com/kubernetes/kubernetes/pull/64662), this PR does note provide e2e test due to concerns about a large image may add too much overhead to the testing infrastructure and pipeline. We should add e2e tests in the future with the use of large enough image(s) in following PRs. 

- Compared to [#64662](https://github.com/kubernetes/kubernetes/pull/64662), this PR simplifies the code changes and keeps code changes under test/integration/scheduler/.

- The PR contains a bug fix for [#65745](https://github.com/kubernetes/kubernetes/pull/65745) - caught by the integration test - where the image states are not properly cloned to the scheduler's cachedNodeInfoMap. We might split this fix into a separate PR.

The integration test covers what follows: a pod requiring a large image (~= 3GB) is submitted to the cluster and there is a single node in the cluster has the same large image; the pod should get scheduled to that node. We might also consider whether more scenarios are desired.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

Kindly ping @resouer and @bsalamat 

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/scheduler/algorithmprovider/defaults/defaults.go [hunk 1]
```diff
@@ -260,6 +256,11 @@ func defaultPriorities() sets.String {
 
 		// Prioritizes nodes that marked with taint which pod can tolerate.
 		factory.RegisterPriorityFunction2("TaintTolerationPriority", priorities.ComputeTaintTolerationPriorityMap, priorities.ComputeTaintTolerationPriorityReduce, 1),
+
+		// ImageLocalityPriority prioritizes nodes based on locality of images requested by a pod. Nodes with larger size
+		// of already-installed packages required by the pod will be preferred over nodes with no already-installed
```
Comment (resouer) [inline pkg/scheduler/algorithmprovider/defaults/defaults.go]: I just feel this comment is too verbose, how about stop at "pod will be preferred."?

Comment (silveryfu) [reply pkg/scheduler/algorithmprovider/defaults/defaults.go]: Sure - let me rephrase it.

File: test/integration/scheduler/priorities_test.go [hunk 2]
```diff
@@ -172,3 +173,66 @@ func TestPodAffinity(t *testing.T) {
...
+
+	// Add a few nodes.
+	_, err := createNodes(context.clientSet, "testnode", nil, 10)
+	if err != nil {
+		t.Fatalf("cannot create nodes: %v", err)
+	}
+
+	// We use a fake large image as the test image used by the pod, which has relatively large image size.
+	// TODO: replace this image with a LargeImage dedicated to testing image locality once the LargeImage is added
+	// TODO: change the image name to the CRI compliant name
```
[the remark this answers is not in the document]
Comment (silveryfu) [reply test/integration/scheduler/priorities_test.go]: :) I will just remove this TODO as it can be misleading - the name used here is fake-large-image:v1 which might be expanded to say docker.io/fake-large-image:v1 - but since the current matching works for both cases, I use the former for simplicity.

**Comment by silveryfu** [comment]
Nits addressed - thanks for the comments @resouer 👍

File: test/integration/scheduler/priorities_test.go [hunk 3]
```diff
@@ -172,3 +173,65 @@ func TestPodAffinity(t *testing.T) {
...
+	defer cleanupTest(t, context)
+
+	// Add a few nodes.
+	_, err := createNodes(context.clientSet, "testnode", nil, 10)
+	if err != nil {
+		t.Fatalf("cannot create nodes: %v", err)
+	}
+
+	// We use a fake large image as the test image used by the pod, which has relatively large image size.
+	// TODO: replace this image with a LargeImage dedicated to testing image locality once the LargeImage is added
```
Comment (resouer) [inline test/integration/scheduler/priorities_test.go]: Are your suggesting using a large e2e test image here? I think in integration test, it seems we will always have to use fake image like this.

Comment (silveryfu) [reply test/integration/scheduler/priorities_test.go]: I agree. The intention was that if there were a large-image in the e2e images (imageutils) then we would be able to simply use it - but this does not justify a TODO.

It will be removed. Thanks!

**Comment by resouer** [comment]
Thanks, this feature is on track for 1.12 now, maybe need to apply some labels? cc @dims

**Comment by dims** [comment]
/milestone v1.12

per last comment from @resouer
